packages/next/build/webpack/loaders/next-image-loader.js

```diff
@@ -0,0 +1,54 @@
+import loaderUtils from 'next/dist/compiled/loader-utils'
+import sizeOf from 'image-size'
+import { processBuffer } from '../../../next-server/server/lib/squoosh/main'
+import fs from 'fs'
+
+const PLACEHOLDER_SIZE = 6
+
+async function nextImageLoader(content) {
+  const context = this.rootContext
+  const opts = { context, content }
+  const interpolatedName = loaderUtils.interpolateName(
+    this,
+    '/static/image/[path][name].[hash].[ext]',
+    opts
+  )
+
+  let extension = loaderUtils.interpolateName(this, '[ext]', opts)
```

Comment:
Can't we use https://www.npmjs.com/package/file-type instead, this ensures the content is always the correct type.